sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -225,10 +332,18 @@ public <T> Mono<CosmosItemResponse<Object>> deleteItem(T item, CosmosItemRequest
      */
     // TODO Make this api public once it is GA in cosmos core library
     Mono<CosmosItemResponse<Object>> deleteAllItemsByPartitionKey(PartitionKey partitionKey, CosmosItemRequestOptions requestOptions) {
-        if (requestOptions == null) {
-            requestOptions = new CosmosItemRequestOptions();
-        }
-        return container.deleteAllItemsByPartitionKey(partitionKey, requestOptions);
+        Objects.requireNonNull(partitionKey, "partitionKey cannot be null");
```

Comment:
Adding a new exception could be breaking. Was the same exception being thrown before as well (just deeper in one of the called methods)?